db/migrate/20181113182615_add_vms_table.rb

```diff
@@ -0,0 +1,26 @@
+class AddVmsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :vms do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref
+      t.string :uuid
```

Comment:
will this be external_uuid or internal?

Reply:
External, thinking the replacement for uid_ems.  Might just call it uid_ems, trying to keep the naming closer to MIQ than we've been doing.

Reply:
right, we've talked about this. Ok I am fine with :uid_ems, or :external_uuid

We might have cases where compose ems_ref from multiple attributes though, so maybe this could just be in :extra column?

Reply:
I think we'll use this for cross-linking right?  I can name it uid_ems but I think having it separate is a good idea since this is a common concept.